zcash_client_backend/src/tor/http/exchange_rate.rs

```diff
@@ -0,0 +1,373 @@
+use futures_util::join;
+use hyper::StatusCode;
+use iso_currency::Currency;
+use rand::{seq::IteratorRandom, thread_rng};
+use rust_decimal::Decimal;
+use serde::Deserialize;
+use tracing::{error, trace};
+
+use crate::tor::{Client, Error};
+
+impl Client {
+    /// Fetches the latest `currency`/ZEC exchange rate, derived from several exchanges.
+    ///
+    /// Supported currencies:
+    /// - USD
+    ///
+    /// Returns:
+    /// - `Ok(Some(rate))` if at least one exchange request succeeds.
+    /// - `Ok(None)` if the given currency is unsupported.
+    /// - `Err(_)` if none of the exchange queries succeed.
+    pub async fn get_exchange_rate(&self, currency: Currency) -> Result<Option<Decimal>, Error> {
```

Comment:
Hmm, now that I'm always fetching `USD(T)/ZEC`, maybe this API should go back to being single-currency:
```suggestion
    pub async fn get_latest_zec_to_usd_rate(&self) -> Result<Decimal, Error> {
```
and then if the caller wants something other than `USD/ZEC`, it can join this itself with a call to a (to-be-implemented) `get_latest_usd_rate(&self, currency: Currency)` function? That would keep the inner implementation of this method simpler.

Reply:
@nuttycom I have this change implemented locally; if you agree with it, I'll force-push the last commit.

Reply:
I'm somewhat indifferent, as I don't think that this should be a concrete method on `Client` but should instead be a trait method that allows others to implement their own preferences for what data source(s) to connect to. IMO without that sort of extensibility available to the third-party users of the crate, this functionality is basically a proof of concept; anyone wishing to use it will likely end up forking.